propagation_test.go

```diff
@@ -48,8 +48,8 @@ func TestSpanPropagator(t *testing.T) {
 		typ, carrier interface{}
 	}{
 		{basictracer.Delegator, basictracer.DelegatingCarrier(&verbatimCarrier{b: map[string]string{}})},
-		{opentracing.SplitBinary, opentracing.NewSplitBinaryCarrier()},
-		{opentracing.SplitText, opentracing.NewSplitTextCarrier()},
+		{opentracing.Binary, opentracing.BinaryCarrier(&[]byte{})},
```

Comment:
A test should be added for `opentracing.BinaryCarrier(nil)`. 

`opentracing.BinaryCarrier` now specifies that "It is fine to pass `nil` to Inject(); in that case, it will allocate a []byte for the caller.", so the test would be checking that behavior.